### ISSUE TITLE
Flying in the air prior to joining parkour mode provides you with a slight super jump before it detects you are flying.

### ISSUE THREAD

**Issue by mibby** [A5H73Y/Parkour#43]
Compiled as of https://github.com/A5H73Y/Parkour/commit/d3cbb66f46a7bfcc1755b11293a3769b999f9f7d
Related https://github.com/A5H73Y/Parkour/issues/28

When you double-jump after joining parkour mode while flying in air prior, you get an initial boost from the flying until it then disables the flight state. Would it be possible to force disable flight when joining parkour mode rather than wait until it detects a player is trying to fly? So they don't get an initial double-jump super-jump.

**Comment by A5H73Y** [comment]
Not entirely sure of how to recreate this.
Is the player survival when they join a course? 
I've done a minor change to disallow flying when you join a course, but it already disabled flying if you already were?

**Comment by mibby** [comment]
Yes, it is when a player has flight mode enabled and they join when in survival mode, flying in the air prior. Not joining from a ground position. It is as if flight mode is still enabled when they get put in parkour and they are allowed to fly up a block in height before it detects and disables.
